packages/medusa/src/interfaces/transaction-base-service.ts

```diff
@@ -2,14 +2,20 @@ import { EntityManager } from "typeorm"
 import { IsolationLevel } from "typeorm/driver/types/IsolationLevel"
 
 export abstract class TransactionBaseService {
-  protected abstract manager_: EntityManager
-  protected abstract transactionManager_: EntityManager | undefined
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+
+  protected get activeManager_(): EntityManager {
+    return this.transactionManager_ ?? this.manager_
+  }
 
   protected constructor(
     protected readonly __container__: any,
     protected readonly __configModule__?: Record<string, unknown>,
     protected readonly __moduleDeclaration__?: Record<string, unknown>
-  ) {}
+  ) {
+    this.manager_ = __container__.manager
```

Comment:
question: Can this have unexpected side effects? Seems to be unrelated to the general boy scouting in the PR

Reply:
it should not have unexpected side effect since that if you want to use the TransctionBaseService the manager should be present. The transactionBaseService itself use the manager_ so it expect it to be present. Also, the property is not optional so it has to be populated.